pkg/images/utils/utils.go

```diff
@@ -264,3 +278,27 @@ func FilterSuppressedCVEsNoClone(img *storage.Image) {
 		}
 	}
 }
+
+// IsRedHatImage takes in an image and returns whether it's a Red Hat image.
+// This is determined via heuristics, by looking at these images, which are assumed to be "official Red Hat images",
+// and checking where they are hosted:
+//
+//   - All images running in a default openshift cluster
+//   - All images that are referred by PackageManifests in the "redhat-operators" OLM catalog
+//
+// See the description of https://github.com/stackrox/stackrox/pull/15761 for details
+// This function is used to determine which images should be built and signed by Red Hat, for supply chain provenance
+func IsRedHatImage(img *storage.Image) bool {
+	// First consider registries where all images are built by Red Hat
+	imageRegistry := img.GetName().GetRegistry()
```

Comment:
Do we need to check Names instead?

Reply:
Yes! done here: https://github.com/stackrox/stackrox/pull/15761/commits/72dd89d5dbdbdd49342bbf2250e54cac3b9f694d